### ISSUE TITLE
blacklists are not working

### ISSUE THREAD

**Issue by dotw** [istio/istio#12987]
**Describe the bug**
Black list works in istio 1.0.5 but doesn't work in 1.1.1

Here is my settings:
``` yaml
---
apiVersion: v1
items:
- apiVersion: config.istio.io/v1alpha2
  kind: listentry
  metadata:
    creationTimestamp: "2019-04-02T08:22:09Z"
    generation: 1
    name: details-hzqi-test-version-listentry
    namespace: hzqi-test
    resourceVersion: "13877651"
    selfLink: /apis/config.istio.io/v1alpha2/namespaces/hzqi-test/listentries/details-hzqi-test-version-listentry
    uid: 68c4987e-5520-11e9-a25f-005056a45a16
  spec:
    value: source.labels["app"] + "." + source.namespace + "-version:" + source.labels["version"]
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
---
apiVersion: v1
items:
- apiVersion: config.istio.io/v1alpha2
  kind: listchecker
  metadata:
    creationTimestamp: "2019-04-02T08:22:09Z"
    generation: 1
    name: details-hzqi-test-black-listchecker
    namespace: hzqi-test
    resourceVersion: "13877650"
    selfLink: /apis/config.istio.io/v1alpha2/namespaces/hzqi-test/listcheckers/details-hzqi-test-black-listchecker
    uid: 68c10d20-5520-11e9-a25f-005056a45a16
  spec:
    blacklist: true
    cachingInterval: 0s
    overrides:
    - 'productpage.hzqi-test-version: v1'
    ttl: 0s
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
---
apiVersion: v1
items:
- apiVersion: config.istio.io/v1alpha2
  kind: rule
  metadata:
    creationTimestamp: "2019-04-02T08:22:11Z"
    generation: 1
    name: details-hzqi-test-black-rule
    namespace: hzqi-test
    resourceVersion: "13877659"
    selfLink: /apis/config.istio.io/v1alpha2/namespaces/hzqi-test/rules/details-hzqi-test-black-rule
    uid: 69ea545c-5520-11e9-a25f-005056a45a16
  spec:
    actions:
    - handler: details-hzqi-test-black-listchecker.listchecker
      instances:
      - details-hzqi-test-version-listentry.listentry
    match: destination.labels["app"] == "details" && destination.namespace == "hzqi-test"
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
---

```

**Expected behavior**
blacklist should be work in istio 1.1.1

**Steps to reproduce the bug**
deploy the bookinfo sample into namespace hzqi-test and then create resources provided upper.

**Version**
istio 1.1.1
kube 1.13.3

**Installation**
using helm chart

**Comment by bianpengyuan** [comment]
In istio 1.1 we disabled policy enforcement by default. Did you enable policy enforcement when you run your test? https://istio.io/docs/tasks/policy-enforcement/enabling-policy/

**Comment by dotw** [comment]
@bianpengyuan thanks. I'll check the settings and feedback.

**Comment by dotw** [comment]
Hi, I confirm the blacklist works when enabled the policy following the doc, but it doesn't work as 1.0.5 even I set `cachingInterval: 0s` and `ttl: 0s`.  The details service is blocked when the rule is created immediately. this is what we expect, but the details service is still blocked event I deleted the rule.  When I refresh the bookinfo page frequently, I can see the details several times. this is not what we expect. We want to see the details page always if the rule doesn't exist.
